ssc/cmod_pvwattsv5.cpp

```diff
@@ -265,338 +263,339 @@ class cm_pvwattsv5_base : public compute_module
-            wdprov = std::unique_ptr<weather_data_provider>(new weatherfile(file));
-
-            weatherfile *wfile = dynamic_cast<weatherfile*>(wdprov.get());
-            if (!wfile->ok()) throw exec_error("pvwattsv5", wfile->message());
-            if (wfile->has_message()) log(wfile->message(), SSC_WARNING);
-        }
-        else if (is_assigned("solar_resource_data"))
-        {
-            wdprov = std::unique_ptr<weather_data_provider>(new weatherdata(lookup("solar_resource_data")));
-        }
-        else
-            throw exec_error("pvwattsv5", "no weather data supplied");
-
-        setup_system_inputs(); // setup all basic system specifications
-
-        adjustment_factors haf(this, "adjust");
-        if (!haf.setup())
-            throw exec_error("pvwattsv5", "failed to setup adjustment factors: " + haf.error());
-
-        // read all the shading input data and calculate the hourly factors for use subsequently
-        shading_factor_calculator shad;
-        if (!shad.setup(this, ""))
-            throw exec_error("pvwattsv5", shad.get_error());
-
-        weather_header hdr;
-        wdprov->header(&hdr);
-
-        // assumes instantaneous values, unless hourly file with no minute column specified
-        double ts_shift_hours = 0.0;
-        bool instantaneous = true;
-        if (wdprov->has_data_column(weather_data_provider::MINUTE))
-        {
-            // if we have an file with a minute column, then
-            // the starting time offset equals the time
-            // of the first record (for correct plotting)
-            // this holds true even for hourly data with a minute column
-            weather_record rec;
-            if (wdprov->read(&rec))
-                ts_shift_hours = rec.minute / 60.0;
-
-            wdprov->rewind();
-        }
-        else if (wdprov->nrecords() == 8760)
-        {
-            // hourly file with no minute data column.  assume
-            // integrated/averaged values and use mid point convention for interpreting results
-            instantaneous = false;
-            ts_shift_hours = 0.5;
-        }
-        else
-            throw exec_error("pvwattsv5", "subhourly weather files must specify the minute for each record");
-
-        assign("ts_shift_hours", var_data((ssc_number_t)ts_shift_hours));
-
-        weather_record wf;
-
-        size_t nyears = 1;
-        std::vector<double> degradationFactor;
-        if (as_boolean("system_use_lifetime_output")) {
-            nyears = as_unsigned_long("analysis_period");
-            std::vector<double> ac_degradation = as_vector_double("dc_degradation");
-            if (ac_degradation.size() == 1) {
-                degradationFactor.push_back(1.0);
-                for (size_t y = 1; y < nyears; y++) {
-                    degradationFactor.push_back(pow((1.0 - ac_degradation[0] / 100.0), y));
-                }
-            }
-            else {
-                for (size_t y = 0; y < nyears; y++) {
```

Comment:
there needs to be a check on the length of the ac_degradation vector before indexing it from 0 to nyears.

Reply:
True! Added to both pvwattsv5 and v7.